index.d.ts

```diff
@@ -4,23 +4,23 @@ import {
   Handler
 } from 'aws-lambda';
 
-declare type EventType<T> =
-  T extends (event: infer EventArgType, context: Context, callback: Callback<any>) => void ? EventArgType :
-  T extends (event: infer EventArgType, context: Context) => Promise<any> ? EventArgType :
+declare type EventType<T, C> =
```

Comment:
From the height of my Typescript *noobness*: does it make sense here to say `C extends Context`?

And same in all the other instances below...

Reply:
This type should be already inherited from function that started it, `const middy` in this case. In here this is just generic type that was defined before and is just passed along.
I've tested it on my code and if you don't define custom Context, the generic one is used

Reply:
@sdomagala is right, line 21 already declares `C extends Context = Context`